server/events/runtime/run_step_runner_test.go

```diff
@@ -87,6 +87,7 @@ func TestRunStepRunner_Run(t *testing.T) {
 		Workspace:        "myworkspace",
 		RepoRelDir:       "mydir",
 		TerraformVersion: projVersion,
+		ProjectName:      "myproject",
```

Comment:
Because `ProjectName` isn't always set, I'd like to test both cases. Can you make `ProjectName` part of the `cases` anonymous struct:
```go
	cases := []struct {
		Command string
                 ProjectName string
		ExpOut  string
		ExpErr  string
	}{
```
(you'll also have to move the initialization of `ctx` to inside the `t.Run()` func)

And then you can test the case when it is and isn't set.

Reply:
I fully understand what you're looking for, and I (kind of) managed to implement it as well, but the implementation is not too DRY (two set of cases, two context declarations, and two loops to execute the tests), since my experience of programming, especially in golang, is closer to 0 than 100.

Reply:
> I fully understand what you're looking for, and I (kind of) managed to implement it as well, but the implementation is not too DRY (two set of cases, two context declarations, and two loops to execute the tests), since my experience of programming, especially in golang, is closer to 0 than 100.

All good, I've made a small change to your work here: https://github.com/runatlantis/atlantis/pull/578 and will merge it from there